Add IndexedDB Extension

ffghdfh opened the pull request:
lemme in please 🥺

ffghdfh committed:
Create indexedDB.js

ffghdfh committed:
Update indexedDB.js license

ffghdfh committed:
fixed dumb issues

ffghdfh committed:
Create indexedDB.svg

ffghdfh committed:
Update extensions.json

ffghdfh committed:
made indexedDB.svg's background non transparent

ffghdfh committed:
oops, lowercase i in extensions.json

ffghdfh committed:
Update README.md, added indexedDB.svg attribution

ffghdfh committed:
Update README.md, forgot to add my name and a /, i am so sorry

ffghdfh committed:
Update indexedDB.js, removed all the reject()-s

ffghdfh committed:
made indexedDB.js prettier :)

grawr

ffghdfh committed:
Update indexedDB.js with discord username

ffghdfh committed:
Update indexedDB.js oops forgot punctuation

ffghdfh committed:
Update indexedDB.js, even MORE fixes

CubesterYT commented:
Please don't close and open the PRs. Are you experiencing some kind of problem?

ffghdfh commented:
> Please don't close and open the PRs. Are you experiencing some kind of problem?

Oh sh#t sorry, didn't know I shouldn't do that😅
i've just been trying to add my extension and there was always something that came up! But i guess i just gotta play the waiting game...

CST1229 commented:
> i've just been trying to add my extension and there was always something that came up! But i guess i just gotta play the waiting game...

Don't expect pull requests to be merged soon; GarboMuffin (TurboWarp's creator) has a job and mostly only is active here on weekends.

ffghdfh commented:
> > i've just been trying to add my extension and there was always something that came up! But i guess i just gotta play the waiting game...
> 
> Don't expect pull requests to be merged soon; GarboMuffin (TurboWarp's creator) has a job and mostly only is active here on weekends.

OHHH! I THOUGHT ALL OF Y'ALL WHO COMMENT COULD TOTALLY MERGE IT AND JUST DIDN'T WANT TO 😅
Okay that's good to know, thank you for letting me know about that too! :D

ffghdfh committed:
Update indexedDB.js, added // By: and set undefined key resolve to just ""

ffghdfh commented on extensions/ffghdfh/indexedDB.js:
```diff
@@ -0,0 +1,225 @@
+// Name: Indexed Database
+// ID: ffghdfhIndexedDB
+// Description: Lets you use an IndexedDB database, allowing for you to store more than 5MB of data locally. By chair_senpai.
```
Okay so i think you probably meant the // By: thingy, sorry for leaving that out, I just went off of the local storage extension which did not have that, oops!

ffghdfh commented on extensions/ffghdfh/indexedDB.js:
```diff
@@ -0,0 +1,225 @@
+
+    getValue(args) {
+      const { KEY } = args;
+      return this._getValue(KEY);
```
Okie dokie, changed that too!

ffghdfh reviewed (commented):
Changed what y'all asked for!

ffghdfh committed:
oh my god i missed the part about scratch profile links only. changed that

DNin01 commented:
I worry that projects might abuse this to take up lots of storage on users' devices, and local storage is fine for the most common use cases. Does IndexedDB have any limits?

CST1229 commented:
> I worry that projects might abuse this to take up lots of storage on users' devices, and local storage is fine for the most common use cases. Does IndexedDB have any limits?

https://developer.mozilla.org/en-US/docs/Web/API/Storage_API/Storage_quotas_and_eviction_criteria#other_web_technologies
there is a limit, but it's gigabytes in size (and also browser-dependent)

ffghdfh committed:
Merge branch 'TurboWarp:master' into IndexedDB-Extension

ffghdfh committed:
Merge branch 'TurboWarp:master' into IndexedDB-Extension

ffghdfh committed:
Merge branch 'TurboWarp:master' into IndexedDB-Extension

ffghdfh committed:
add new features and improve the extension

Updated IndexedDB extension with new features and improvements

- Renamed "get value for key ()" to "get key ()"
- Set all default block values to empty strings
- Added "export db" reporter block to export the entire database as a JSON string
- Added "import db from ()" command block to import and replace the database from a JSON string
- Added "merge db from ()" command block to merge data from a JSON string without overwriting existing keys

ffghdfh committed:
Removed monitor checkbox

ffghdfh committed:
Made get key return nothing

Made the "get key" block return nothing because this extension won't ever be merged into the turbowarp extensions anyways so I don't care anymore lol imma do it my way

CubesterYT commented:
Hey man! Remember me? I'm a TurboWarp Collaborater now so I can do a full thorough review and revisions on your PR, such as prettier, translation functions, etc. So let me know if your PR is ready and I'll do the things.

ffghdfh committed:
Merge branch 'TurboWarp:master' into IndexedDB-Extension

ffghdfh committed:
Merge branch 'TurboWarp:master' into IndexedDB-Extension

Thebloxers998 reviewed (approved):
I've been looking for a indexedDB extension

Drago-Cuven commented:
miyo made a better one.

BlockBoi19 commented:
> miyo made a better one.

???

Thebloxers998 commented:
> miyo made a better one.

Link?

Thebloxers998 commented on extensions/ffghdfh/indexedDB.js:
```diff
@@ -0,0 +1,316 @@
+(function (Scratch) {
```
We really need this because it's important

ffghdfh reviewed (commented):
Lol this all makes sense and stuff thanks guys lol
btw i'm not even sure this is the version i use since i've been using the one I sumbitted here https://github.com/ffghdfh/lemons-gallery-ff/blob/main/extensions/golbo/IndexedDB.js

ffghdfh committed:
Merge branch 'TurboWarp:master' into IndexedDB-Extension

CubesterYT commented:
!format

GarboMuffin commented:
!format

DangoCat committed:
[Automated] Format code

CubesterYT commented:
It worked!

Thebloxers998 commented:
> It worked!

Yep, it did

ffghdfh committed:
Merge branch 'TurboWarp:master' into IndexedDB-Extension

GarboMuffin commented:
I have been stalling on this because the dream is that users don't need to worry about where the data is stored; we transparently put it wherever it makes sense to.

ffghdfh committed:
Merge branch 'TurboWarp:master' into IndexedDB-Extension

Brackets-Coder commented:
@ffghdfh is this still being worked on?

ffghdfh commented:
> @ffghdfh is this still being worked on?

Nnnnope not really, it works fine enough I guess? feel free to poke at it if you'd like

ffghdfh commented:
> > i've just been trying to add my extension and there was always something that came up! But i guess i just gotta play the waiting game...
> 
> Don't expect pull requests to be merged soon; GarboMuffin (TurboWarp's creator) has a job and mostly only is active here on weekends.

Mind you this was in 2024 and it still has not been merged yet

Brackets-Coder commented:
> Mind you this was in 2024 and it still has not been merged yet

Yes, I know this repository is notoriously slow. I'm doing my best to increase merge rates but to be honest we don't have the frameworks to keep the tap water flowing.

As garbomuffin mentioned, any kind of "storage" extension becomes complicated, so I'll see what I can do to review later (I'm busy right now). Hopefully we can get this merged soon but I really can't say for sure :/

ffghdfh commented:
That's alright, thank you for your help though :D

On Sun, 22 Feb 2026 at 19:14, Brackets-Coder ***@***.***>
wrote:

> *Brackets-Coder* left a comment (TurboWarp/extensions#1341)
> <https://github.com/TurboWarp/extensions/pull/1341#issuecomment-3941471458>
>
> Mind you this was in 2024 and it still has not been merged yet
>
> Yes, I know this repository is notoriously slow. I'm doing my best to
> increase merge rates but to be honest we don't have the frameworks to keep
> the tap water flowing.
>
> As garbomuffin mentioned, any kind of "storage" extension becomes
> complicated, so I'll see what I can do to review later (I'm busy right
> now). Hopefully we can get this merged soon but I really can't say for sure
> :/
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/TurboWarp/extensions/pull/1341#issuecomment-3941471458>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AW7BSMRCVITIBP5452AB3MT4NHWZVAVCNFSM6AAAAACVG7CWUSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTSNBRGQ3TCNBVHA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

Brackets-Coder commented:
You mean like do a IndexedDB API backend for Local Storage?
